wpilibc/src/main/native/cpp/spline/SplineHelper.cpp

```diff
@@ -99,38 +82,68 @@ std::vector<CubicHermiteSpline> SplineHelper::CubicSplinesFromWaypoints(
       splines.push_back(spline);
     }
   } else if (waypoints.size() == 1) {
-    const double xDeriv = (3 * (end.Translation().X().to<double>() -
-                                start.Translation().X().to<double>()) -
-                           endXControlVector[1] - startXControlVector[1]) /
-                          4.0;
-    const double yDeriv = (3 * (end.Translation().Y().to<double>() -
-                                start.Translation().Y().to<double>()) -
-                           endYControlVector[1] - startYControlVector[1]) /
-                          4.0;
+    const double xDeriv =
+        (3 * (xFinal[0] - xInitial[0]) - xFinal[1] - xInitial[1]) / 4.0;
+    const double yDeriv =
+        (3 * (yFinal[0] - yInitial[0]) - yFinal[1] - yInitial[1]) / 4.0;
 
     std::array<double, 2> midXControlVector{waypoints[0].X().to<double>(),
                                             xDeriv};
     std::array<double, 2> midYControlVector{waypoints[0].Y().to<double>(),
                                             yDeriv};
 
-    splines.emplace_back(startXControlVector, midXControlVector,
-                         startYControlVector, midYControlVector);
-    splines.emplace_back(midXControlVector, endXControlVector,
-                         midYControlVector, endYControlVector);
+    splines.emplace_back(xInitial, midXControlVector, yInitial,
+                         midYControlVector);
+    splines.emplace_back(midXControlVector, xFinal, midYControlVector, yFinal);
 
   } else {
     // Create the spline.
-    const CubicHermiteSpline spline{startXControlVector, endXControlVector,
-                                    startYControlVector, endYControlVector};
+    const CubicHermiteSpline spline{xInitial, xFinal, yInitial, yFinal};
     splines.push_back(spline);
   }
 
   return splines;
 }
 
-std::vector<QuinticHermiteSpline> SplineHelper::QuinticSplinesFromWaypoints(
-    const std::vector<Pose2d>& waypoints) {
+std::vector<QuinticHermiteSpline>
+SplineHelper::QuinticSplinesFromControlVectors(
+    const std::vector<Spline<5>::ControlVector>& controlVectors) {
   std::vector<QuinticHermiteSpline> splines;
+  for (unsigned int i = 0; i < controlVectors.size() - 1; i++) {
```

Comment:
Array indices typically use size_t. Also, should use ++i, but the optimizer probably notices the dead write anyway and removes it.